jsonrpc/server_test.go

```diff
@@ -0,0 +1,147 @@
+package jsonrpc_test
+
+import (
+	"testing"
+
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServer_RegisterMethod(t *testing.T) {
+	server := jsonrpc.NewServer()
+	tests := map[string]struct {
+		handler    any
+		paramNames []jsonrpc.Parameter
+		want       string
+	}{
+		"not a func handler": {
+			handler: 44,
+			want:    "handler must be a function",
+		},
+		"excess param names": {
+			handler:    func() {},
+			paramNames: []jsonrpc.Parameter{{Name: "param1"}},
+			want:       "number of function params and param names must match",
+		},
+		"missing param names": {
+			handler:    func(param1, param2 int) {},
+			paramNames: []jsonrpc.Parameter{{Name: "param1"}},
+			want:       "number of function params and param names must match",
+		},
+		"no return": {
+			handler: func(param1, param2 int) {},
+			want:    "handler must return 2 values",
+		},
+		"int return": {
+			handler: func(param1, param2 int) (int, int) { return 0, 0 },
+			want:    "second return value must be a *jsonrpc.Error",
+		},
+		"no error return": {
+			handler: func(param1, param2 int) (any, int) { return 0, 0 },
+			want:    "second return value must be a *jsonrpc.Error",
+		},
+	}
+
+	for desc, test := range tests {
+		err := server.RegisterMethod("method", test.paramNames, test.handler)
+		assert.EqualError(t, err, test.want, desc)
+	}
+
+	err := server.RegisterMethod("method", nil, func(param1, param2 int) (int, *jsonrpc.Error) { return 0, nil })
+	assert.NoError(t, err)
+}
+
+func TestHandle(t *testing.T) {
+	server := jsonrpc.NewServer()
+	require.NoError(t, server.RegisterMethod("method", []jsonrpc.Parameter{{Name: "num"}, {Name: "shouldError", Optional: true}, {Name: "msg", Optional: true}},
+		func(num *int, shouldError bool, msg string) (any, *jsonrpc.Error) {
+			if shouldError {
+				return nil, &jsonrpc.Error{Code: 44, Message: msg}
+			}
+			return struct {
+				Doubled int `json:"doubled"`
+			}{*num * 2}, nil
+		}))
+
+	tests := []struct {
+		req string
+		res string
+	}{
+		{
+			req: `{]`,
+			res: `{"jsonrpc":"2.0","error":{"code":-32600,"message":"invalid character ']' looking for beginning of object key string"},"id":null}`,
```

Comment:
This test case should return `-32700` which is for invalid json. `-32600` is for well-formed json but an invalid request.

Reply:
yeap, fixed